src/Field.php

```diff
@@ -508,6 +509,36 @@ public function useAlias(): bool
 
     // }}}
 
+    // {{{ Scope condition
+
+    /**
+     * Returns arguments to be used for query on this field based on the condition.
+     *
+     * @param string|null $operator one of Scope\Condition operators
+     * @param mixed       $value    the condition value to be handled
+     *
+     * @todo: value is array
```

Comment:
why do we need to support an array?

Reply:
We do support array but we pass it onto dsql to handle it converting to `value in (a, b, c)`
The problem is if we have an array as value in Condition we need to typecast each value for correct results in dsql query
This is clearly the solution when we have '=' as operator.
But what to do if we have `new Condition($field, '>', [a, b, c])`?

Reply:
`> arr` may seems like AND expand, but I think we should not support it

so like you have `$skipValueTypecast` we should have `$valueCanBeArray` with `=` and `!=` operators and typecast individually. I belive, that we should still support only single level array and throw if more than that or operator does not support array.

Reply:
We are thinking in same direction - validate operator-value combination when constructing the condition
- accept array as value only when = and != operators
- accept only single level arrays

Reply:
> validate operator-value combination when constructing the condition

👍

Reply:
Done. Let me know if some comments so i can then merge